File: app/src/main/java/com/justnik/mirea_navigation_lesson/ui/home/DogFragment.java
package com.justnik.mirea_navigation_lesson.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import com.justnik.mirea_navigation_lesson.R;
import com.justnik.mirea_navigation_lesson.databinding.FragmentDogBinding;


public class DogFragment extends Fragment {

    private FragmentDogBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentDogBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        binding.bDog.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).popBackStack(R.id.homeFragment, false);
            }
        });
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        binding = null;
    }
}

File: app/src/main/java/com/justnik/mirea_navigation_lesson/ui/home/DogFragment.kt
package com.justnik.mirea_navigation_lesson.ui.home

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import by.kirich1409.viewbindingdelegate.viewBinding
import com.justnik.mirea_navigation_lesson.R
import com.justnik.mirea_navigation_lesson.databinding.FragmentDogBinding

class DogFragment : Fragment(R.layout.fragment_dog) {
    private val binding: FragmentDogBinding by viewBinding()


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        setClickListener()
    }

    private fun setClickListener() {
        binding.bDog.setOnClickListener {
            findNavController().popBackStack(R.id.homeFragment, false)
        }
    }
}